support-models/src/main/scala/com/gu/support/workers/PaymentFields.scala

```diff
@@ -1,18 +1,40 @@
 package com.gu.support.workers
 
+import cats.syntax.functor._
 import com.gu.support.encoding.Codec
 import com.gu.support.encoding.Codec.deriveCodec
-import io.circe.{Decoder, Encoder}
+import io.circe.{Encoder, _}
 import io.circe.syntax._
-import cats.syntax.functor._
 
 sealed trait PaymentFields
 
+
 case class PayPalPaymentFields(baid: String) extends PaymentFields
 
+
 sealed trait StripePaymentFields extends PaymentFields
 case class StripeSourcePaymentFields(stripeToken: String) extends StripePaymentFields // pre SCA compatibility
-case class StripePaymentMethodPaymentFields(paymentMethod: String) extends StripePaymentFields
+case class StripePaymentMethodPaymentFields(paymentMethod: PaymentMethodId) extends StripePaymentFields
+
+object PaymentMethodId {
+
+  def apply(value: String): Option[PaymentMethodId] =
+    if (value.nonEmpty &&
+      value.forall { char =>
+        (char >= 'a' && char <= 'z') ||
+          (char >= 'A' && char <= 'Z') ||
+          (char >= '0' && char <= '9') ||
+          char == '_'
+      }) Some(new PaymentMethodId(value)) else None
```

Comment:
the main things we want to avoid are slashes and dots, as we concatenate this into a url and it's easier to validate than try to use a url builder to escape things.  We know that payment method ids are supposed to be fairly inoffensive, so this makes things easier to reason about.